File: kmp/signet/src/commonMain/kotlin/xyz/nirapod/signet/kmp/PublicKey.kt
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileCopyrightText: 2026 Nirapod Labs

package xyz.nirapod.signet.kmp

/**
 * A public key in one of the pinned wire formats. There is no matching
 * private-key accessor anywhere in the surface.
 */
public class PublicKey(
    public val format: Format,
    public val bytes: ByteArray,
) {
    /** The public key wire format. */
    public enum class Format {
        /** Uncompressed X9.63 point: `0x04 || X || Y`, 65 bytes for P-256. */
        rawX962,

        /** DER SubjectPublicKeyInfo, 91 bytes for P-256. */
        spki,
    }
}

File: apple/Sources/SignetCore/PublicKey.swift
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileCopyrightText: 2026 Nirapod Labs

import Foundation

/// A public key in one of the contract's wire formats. `rawX962` is the
/// uncompressed X9.63 point (`0x04 || X || Y`); `spki` is the DER
/// SubjectPublicKeyInfo wrapping that point.
public struct PublicKey: Sendable, Equatable {
    public enum Format: String, Sendable, Equatable, CaseIterable {
        case rawX962
        case spki
    }

    public let format: Format
    public let bytes: Data

    public init(format: Format, bytes: Data) {
        self.format = format
        self.bytes = bytes
    }
}
